### PR TITLE
ONBOARDING: Dialog with character pressed

### PR DESCRIPTION
**Pull request by priettt**
When clicking a character, a dialog shows up with the character's information

Signed-off by: Francisco Prieto francisco.prieto@globant.com

### CODE REVIEW

File: app/src/main/java/com/puzzlebench/cleanMarvelKotlin/presentation/mvp/CharacterPresenter.kt [hunk 1]
```diff
@@ -1,16 +1,45 @@
...
 
     fun init() {
         view.init()
         requestGetCharacters()
+        RxBus.subscribe(view.activity, object : OnCharacterPressedBusObserver() {
+            override fun onEvent(value: OnCharacterPressedBusObserver.OnCharacterPressed) {
+                val subscription = getCharacterDetailServiceUseCase.invoke(value.id).
+                        subscribeOn(Schedulers.io()).observeOn(AndroidSchedulers.mainThread())
+                        .subscribe({ character ->
+                    var detailsLink = ""
```
Comment (SartiMau) [inline app/src/main/java/com/puzzlebench/cleanMarvelKotlin/presentation/mvp/CharacterPresenter.kt]: use constants please

**Review by SartiMau — CHANGES_REQUESTED** [review]

**Comment by SartiMau** [comment]
Please add a video showing how the app is working with the new functionality

File: app/src/main/java/com/puzzlebench/cleanMarvelKotlin/presentation/mvp/CharacterPresenter.kt (hunk 1 above)
Comment (eis6aer) [reply app/src/main/java/com/puzzlebench/cleanMarvelKotlin/presentation/mvp/CharacterPresenter.kt]: I would put the dot on the new line 
```suggestion
                    getCharacterDetailServiceUseCase.invoke(value.id)
                    .subscribeOn(Schedulers.io())
                    .observeOn(AndroidSchedulers.mainThread())
```

**Comment by eis6aer** [comment]
I would put the observable operators on a new line each 
```getCharacterDetailServiceUseCase.invoke(value.id)
                    .subscribeOn(Schedulers.io())
                    .observeOn(AndroidSchedulers.mainThread())

File: app/src/main/java/com/puzzlebench/cleanMarvelKotlin/presentation/mvp/CharacterPresenter.kt (hunk 1 above)
Comment (priettt) [reply app/src/main/java/com/puzzlebench/cleanMarvelKotlin/presentation/mvp/CharacterPresenter.kt]: done!

File: app/src/main/java/com/puzzlebench/cleanMarvelKotlin/presentation/adapter/viewholder/viewholder.kt [hunk 2]
```diff
@@ -13,7 +13,7 @@ class CharactersAdapterViewHolder(view: View, val listener: CharacterListener) :
     fun bind(item: Character) = with(itemView) {
         tv_item.text = item.name
         val string = item.thumbnail.path + "." + item.thumbnail.extension
```
Comment (SartiMau) [inline app/src/main/java/com/puzzlebench/cleanMarvelKotlin/presentation/adapter/viewholder/viewholder.kt]: no use constants and use strings templates pls

File: app/src/main/java/com/puzzlebench/cleanMarvelKotlin/presentation/extension/ImageExtensions.kt [hunk 3]
```diff
@@ -3,6 +3,11 @@ package com.puzzlebench.cleanMarvelKotlin.presentation.extension
 import android.widget.ImageView
 import com.squareup.picasso.Picasso
 
-fun ImageView.getImageByUrl(url: String) {
+fun ImageView.getImageByUrlCenterCrop(url: String) {
     Picasso.get().load(url).fit().centerCrop().into(this)
 }
+
+fun ImageView.getImageByUrl(url: String?) {
+    if (url != null) Picasso.get().load(url).into(this)
```
Comment (SartiMau) [inline app/src/main/java/com/puzzlebench/cleanMarvelKotlin/presentation/extension/ImageExtensions.kt]: use braces and put each function in a separated line pls

**Review by SartiMau — CHANGES_REQUESTED** [review]

**Review by SartiMau — APPROVED** [review]

**Review by SartiMau — APPROVED** [review]

**Review by fanky10 — APPROVED** [review]
LG2M

**Pull request merged**